sycl/test-e2e/format.py

```diff
@@ -342,8 +351,14 @@ def get_extra_env(sycl_devices):
                 elif "level_zero_v1" in full_dev_name:
                     expanded += " env UR_LOADER_USE_LEVEL_ZERO_V2=0"
 
+                # If ZE_AFFINITY_MASK is set, it filters devices so we should use :0
+                device_selector = parsed_dev_name
+                if "ZE_AFFINITY_MASK" in test.config.environment:
+                    backend, _ = parsed_dev_name.split(":", 1)
+                    device_selector = f"{backend}:0"
```

Comment:
@sarnex , should we just unset it and report a warning instead? Somewhere around https://github.com/mateuszpn/llvm/blob/c5fa7266ca7ecfc0a265ba06da9af99a40cfd557/sycl/test-e2e/lit.cfg.py#L77-L96

Reply:
Yeah I recommend this way, having code trying to play nice with existing envvars seems bad